packages/schema/src/addResolversToSchema.ts

```diff
@@ -165,98 +155,96 @@ function addResolversToExistingSchema(
-          } else if (
-            fieldName === 'extensions' &&
-            type.extensions != null &&
-            (resolverValue as GraphQLEnumType).extensions != null
-          ) {
-            type.extensions = Object.assign({}, type.extensions, (resolverValue as GraphQLEnumType).extensions);
-          } else if (enumValueConfigMap[fieldName]) {
-            enumValueConfigMap[fieldName].value = resolverValue[fieldName];
-          }
-        });
+      for (const fieldName in resolverValue) {
+        if (fieldName.startsWith('__')) {
+          config[fieldName.substring(2)] = resolverValue[fieldName];
+        } else if (fieldName === 'astNode' && config.astNode != null) {
+          config.astNode = {
+            ...config.astNode,
+            description: (resolverValue as GraphQLScalarType)?.astNode?.description ?? config.astNode.description,
+            directives: (config.astNode.directives ?? []).concat(
+              (resolverValue as GraphQLEnumType)?.astNode?.directives ?? []
+            ),
+          };
+        } else if (fieldName === 'extensionASTNodes' && config.extensionASTNodes != null) {
+          config.extensionASTNodes = config.extensionASTNodes.concat(
+            (resolverValue as GraphQLEnumType)?.extensionASTNodes ?? []
+          );
+        } else if (
+          fieldName === 'extensions' &&
+          type.extensions != null &&
+          (resolverValue as GraphQLEnumType).extensions != null
+        ) {
+          type.extensions = Object.assign({}, type.extensions, (resolverValue as GraphQLEnumType).extensions);
+        } else if (enumValueConfigMap[fieldName]) {
+          enumValueConfigMap[fieldName].value = resolverValue[fieldName];
+        }
+      }
 
-        typeMap[typeName] = new GraphQLEnumType(config);
-      } else if (isUnionType(type)) {
-        getAllPropertyNames(resolverValue).forEach(fieldName => {
-          if (fieldName.startsWith('__')) {
-            type[fieldName.substring(2)] = resolverValue[fieldName];
-          }
-        });
-      } else if (isObjectType(type) || isInterfaceType(type)) {
-        getAllPropertyNames(resolverValue).forEach(fieldName => {
-          if (fieldName.startsWith('__')) {
-            // this is for isTypeOf and resolveType and all the other stuff.
-            type[fieldName.substring(2)] = resolverValue[fieldName];
-            return;
-          }
+      typeMap[typeName] = new GraphQLEnumType(config);
+    } else if (isUnionType(type)) {
+      for (const fieldName in resolverValue) {
+        if (fieldName.startsWith('__')) {
+          type[fieldName.substring(2)] = resolverValue[fieldName];
+        }
+      }
+    } else if (isObjectType(type) || isInterfaceType(type)) {
+      for (const fieldName in resolverValue) {
+        if (fieldName.startsWith('__')) {
+          // this is for isTypeOf and resolveType and all the other stuff.
+          type[fieldName.substring(2)] = resolverValue[fieldName];
+          break;
```

Comment:
@ardatan I haven't gotten a chance to try a test, but it feels like the two instances of `break` in this file added in this PR should be `continue`. Won't this mean "as soon as you hit a `__` thing, ignore any fields that happen to be after it in resolverValue"? Similarly for the other use.

Reply:
thanks @glasser ! #4355 with fix has been merged by @ardatan  and will be released with next patch release